src/R/Components/Impl/ConnectionManager/Implementation/ConnectionManager.cs

```diff
@@ -60,9 +60,11 @@ internal class ConnectionManager : IConnectionManager {
                 .Add(_statusBarViewModel)
                 .Add(_hostLoadIndicatorViewModel)
                 .Add(() => _sessionProvider.BrokerStateChanged -= BrokerStateChanged)
+                .Add(() => _sessionProvider.SessionStateChanged -= SessionStateChanged)
```

Comment:
`RSessionProvider` in `CreateRSession` assigns same handler to all sessions. I'll try and supply session in the event args